src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1616,10 +1616,7 @@ protected internal override void OnProjectRemoved(ProjectId projectId)
                 base.OnProjectRemoved(projectId);
 
                 _threadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
-                {
-                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-                    RefreshProjectExistsUIContextForLanguage(languageName);
-                });
+                    await RefreshProjectExistsUIContextForLanguage_NoLockAsync(languageName, cancellationToken).ConfigureAwait(false));
```

Comment:
The _NoLock doesn't make sense here -- the RunWithShutdownBlockAsync isn't being awaited, which means the core of the logic is running after the SwitchToMainThreadAsync and the lock isn't being held anymore.